nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -311,7 +311,7 @@ def training_step(self, dataloader_iter, batch_idx):
             dtype=self.autocast_dtype,
             grad_scaler=self.trainer.precision_plugin.scaler.scale if self.cfg.precision == 16 else None,
             sequence_parallel=self.cfg.get('sequence_parallel', False),
-            enable_autocast=True,
+            enable_autocast=False if self.autocast_dtype == torch.float else True,
```

Comment:
This should be False if we're using O2 as well. Maybe just make a flag in the constructor that sets this true only when not using O2 and and when precision is 16 or bf16.

Reply:
okay, but we might have to test that it doesn't break some random niche thing
Previously with Apex, autocast was being set even with O2

Reply:
done!